enhancements/installer/aws-custom-region-and-endpoints.md

```diff
@@ -0,0 +1,321 @@
+
+And in contrast to the current configuration in the `Infrastructure` object like `region` is not editable as day-2 operation, the users should be able to edit the AWS endpoints day-2, therefore this configuration best belongs in the `spec` section.
+
+```go
+// InfrastructureSpec contains settings that apply to the cluster infrastructure.
+type InfrastructureSpec struct {
+  // platformSpec holds configuration specific to the underlying
+  // infrastructure provider.
+  // +optional
+  PlatformSpec *PlatformSpec `json:"platformSpec,omitempty"`
+}
+
+// PlatformSpec holds some configuration to the underlying infrastructure provider
+// of the current cluster.It is supposed that only one of the status structs is set.
+type PlatformSpec struct {
+  // type is the underlying infrastructure provider for the cluster. This
+  // value controls whether infrastructure automation such as service load
+  // balancers, dynamic volume provisioning, machine creation and deletion, and
+  // other integrations are enabled. If None, no infrastructure automation is
+  // enabled. Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
+  // "OpenStack", "VSphere", "oVirt", and "None". Individual components may not support
+  // all platforms, and must handle unrecognized platforms as None if they do
+  // not support that platform.
+  Type PlatformType `json:"type"`
+
+  // AWS contains settings specific to the Amazon Web Services infrastructure provider.
+  // +optional
+  AWS *AWSPlatformSpec `json:"aws,omitempty"`
+
+  // Azure contains settings specific to the Azure infrastructure provider.
+  // +optional
+  Azure *AzurePlatformSpec `json:"azure,omitempty"`
+
+  // GCP contains settings specific to the Google Cloud Platform infrastructure provider.
+  // +optional
+  GCP *GCPPlatformSpec `json:"gcp,omitempty"`
+
+  // BareMetal contains settings specific to the BareMetal platform.
+  // +optional
+  BareMetal *BareMetalPlatformSpec `json:"baremetal,omitempty"`
+
+  // OpenStack contains settings specific to the OpenStack infrastructure provider.
+  // +optional
+  OpenStack *OpenStackPlatformSpec `json:"openstack,omitempty"`
+
+  // Ovirt contains settings specific to the oVirt infrastructure provider.
+  // +optional
+  Ovirt *OvirtPlatformSpec `json:"ovirt,omitempty"`
+}
+
+
+// AWSPlatformSpec holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformSpec struct {
+  // APIEndpoints list contains custom endpoints which will override default
+  // service endpoint of AWS Services.
+  // There must be only one CustomEndpoint for a service.
+  // +optional
+  APIEndpoints []AWSCustomEndpoint `json:"apiEndpoints,omitempty"`
+}
+
+// AWSCustomEndpoint store the configuration of a custom url to
+// override existing defaults of AWS Services.
+// Currently Supports - EC2, IAM, ELB, S3 and Route53.
+type AWSCustomEndpoint struct {
+    Service string `json:"service"`
```

Comment:
Make this a string-based type? Perhaps not since we'd have to update this like crazy after each re:Invent conference...

```
type AWSService string

const (
  IAM AWSService = "IAM"
  EC2 AWSService = "EC2"
  S3 AWSService = "S3"
  ...
)
```